aten/src/ATen/native/native_functions.yaml

```diff
@@ -3819,18 +3819,23 @@
   device_check: NoCheck   # TensorIterator
   structured_delegate: sin.out
   variants: function, method
+  dispatch:
```

Comment:
Every function with `structured_delegate: out variant` needs an explicit formula. In case of `sin` there's one already:
```
- name: sin(Tensor self) -> Tensor
  self: grad * self.cos().conj()
  result: auto_element_wise
```
For now it won't work anyway.

Reply:
Structured doesn't support sparse output at the moment, that's we can't rely on `structured_delegate: sin.out` to generate the function.

Reply:
Ah right, we would need to specify for `sin` in any case. Thanks.